pkiinit/docker-entrypoint.sh

```diff
@@ -23,4 +23,30 @@ set -e
 # all processes in its session. This can alleviate the chance of leaking zombies, 
 # thus more graceful termination of all sub-processes if any.
 
+# create_dir function creates the directory if it is not existing yet
+# it skips the creation if already exists
+create_dir() {
+    local dir_path=$1
+    if  [ ! -d "$dir_path" ]; then 
+      mkdir -p $dir_path
+    fi
+}
+
+# runtime directory is set per user:
+export XDG_RUNTIME_DIR=${XDG_RUNTIME_DIR:-/run/user/$(echo $(id -u))}
+
+PKI_INIT_RUNTIME_DIR=${XDG_RUNTIME_DIR}${PKI_INIT_DIR}
+PKI_INIT_SCRATCH_DIR=${PKI_INIT_RUNTIME_DIR}/scratch
+PKI_INIT_GENERATE_DIR=${PKI_INIT_RUNTIME_DIR}/generated
+
+# debug output:
+echo XDG_RUNTIME_DIR $XDG_RUNTIME_DIR
+echo PKI_INIT_RUNTIME_DIR $PKI_INIT_RUNTIME_DIR
+
+# create scratch directory area
+create_dir ${PKI_INIT_SCRATCH_DIR}
```

Comment:
This is more efficiently written as 
```mkdir -p ${PKI_INIT_SCRATCH_DIR} || true`

Reply:
Bryon, maybe you can educate me here- what does `pipe to true (|| true)` do?

Reply:
Never mind, it is logic or (||) not pipe. read it wrongly

Reply:
here i put the logic in create_dir function and call it with input argument (${PKI_INIT_SCRATCH_DIR} in this case)

Reply:
Ensuring that XDG_RUNTIME_DIR is set for the container if it is not already set is fine.
The executable should create the /scratch and /generated folders since these are private to pkisetup and nobody else needs to know about them.

Reply:
ok